it/uk/gov/hmrc/agentmappingfrontend/controllers/MappingControllerISpec.scala

```diff
@@ -37,42 +36,31 @@ class MappingControllerISpec extends BaseControllerISpec with AuthStubs {
 
     behave like anAuthenticatedEndpoint(GET, "/agent-mapping/start-submit", callEndpointWith)
 
-    "redirect to add-code if the current user is logged in and has legacy agent enrolment for SA" in {
-      givenUserIsAuthenticated(anSAEnrolledAgent)
-      val request = fakeRequest(GET, "/agent-mapping/start-submit")
-      val result = callEndpointWith(request)
-      status(result) shouldBe 303
-      redirectLocation(result).get shouldBe routes.MappingController.showAddCode().url
-    }
-    "redirect to add-code if the current user is logged in and has legacy agent enrolment for VAT" in {
-      givenUserIsAuthenticated(anVATEnrolledAgent)
-      val request = fakeRequest(GET, "/agent-mapping/start-submit")
-      val result = callEndpointWith(request)
-      status(result) shouldBe 303
-      redirectLocation(result).get shouldBe routes.MappingController.showAddCode().url
+    Auth.validEnrolments.foreach { serviceName =>
+      s"redirect to the add-code if the current user is logged in and has legacy agent enrolment for $serviceName" in {
+        givenAuthorisedFor(serviceName)
+        val request = fakeRequest(GET, "/agent-mapping/start-submit")
+        val result = callEndpointWith(request)
+        status(result) shouldBe 303
```

Comment:
line 45, should check whether status is 303

Reply:
should be built into redirectLocation

Reply:
`redirectLocation` is a standard test facility provided by `play.api.test`, it looks at the `Location` header only